presto-spi/src/main/java/com/facebook/presto/spi/ConnectorMaterializedViewDefinition.java

```diff
@@ -0,0 +1,165 @@
+package com.facebook.presto.spi;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Objects.requireNonNull;
+
+public final class ConnectorMaterializedViewDefinition
+{
+    private final String originalSql;
+    private final Optional<String> catalog;
+    private final String schema;
+    private final String table;
+    private final List<ViewBaseTable> baseTables;
+    private final Optional<String> owner;
+    // A 1-to-1 map from view column name to a tuple of base table name + base table column name
+    @JsonIgnore
+    private Optional<Map<String, Map<SchemaTableName, String>>> viewToBaseColumnMap;
+
+    @JsonCreator
+    public ConnectorMaterializedViewDefinition(
+            @JsonProperty("originalSql") String originalSql,
+            @JsonProperty("catalog") Optional<String> catalog,
+            @JsonProperty("schema") String schema,
+            @JsonProperty("table") String table,
+            @JsonProperty("baseTables") List<ViewBaseTable> baseTables,
+            @JsonProperty("owner") Optional<String> owner)
+    {
+        this.originalSql = requireNonNull(originalSql, "originalSql is null");
+        this.catalog = requireNonNull(catalog, "catalog is null");
+        this.schema = requireNonNull(schema, "schema is null");
+        this.table = requireNonNull(table, "table is null");
+        this.baseTables = unmodifiableList(new ArrayList<>(requireNonNull(baseTables, "baseTables is null")));
+        this.owner = requireNonNull(owner, "owner is null");
+        this.viewToBaseColumnMap = Optional.empty(); }
+
+    @JsonProperty
+    public String getOriginalSql()
+    {
+        return originalSql;
+    }
+
+    @JsonProperty
+    public Optional<String> getCatalog()
+    {
+        return catalog;
+    }
+
+    @JsonProperty
+    public String getSchema()
+    {
+        return schema;
+    }
+
+    @JsonProperty
+    public String getTable()
+    {
+        return table;
+    }
+
+    @JsonProperty
+    public List<ViewBaseTable> getBaseTables()
+    {
+        return baseTables;
+    }
+
+    @JsonProperty
+    public Optional<String> getOwner()
+    {
+        return owner;
+    }
+
+    @JsonIgnore
+    public Optional<Map<String, Map<SchemaTableName, String>>> getViewToBaseColumnMap()
+    {
+        return viewToBaseColumnMap;
+    }
+
+    @JsonIgnore
+    public void setViewToBaseColumnMap(Map<String, Map<SchemaTableName, String>> viewToBaseColumnMap)
+    {
+        this.viewToBaseColumnMap = Optional.of(unmodifiableMap(new HashMap<>(requireNonNull(viewToBaseColumnMap, "viewToBaseColumnMap is null"))));
+    }
```

Comment:
I think the following comment on Optional is also related. If we have it always in the struct as well as not use setter, the map will get serialized to Metastore. Then to serialize the json successfully, I need 1) either write a KeySerializer for the class SchemaTableName (https://fburl.com/tq5qcuw1); 2) or use String as the Map key, and add comment about it's a map from what to what.

Just to have us agree on how to handle it before changing everywhere. I prefer to get the map saved to Metastore, so that we could retrieve later at the level of HiveMetastore rather than passing it through from StatementAnalyzer.